packages/snaps-execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -395,7 +395,7 @@ export class BaseSnapExecutor {
       }
     };
 
-    return { request: request as RequestFunction };
+    return proxyStreamProvider(provider, request) as SnapsGlobalObject;
```

Comment:
The snap global does not need to be a EIP 1193 provider. It should just be a hardened proxy that only exposes the request function. 

It is by design that we dont keep using EIP 1193 as we most likely will deprecate it in the future.

Reply:
Thanks for clarification. I've refactored that part for a snap global. New proxy is added, which is now exposing only `request`. Please take a look a again and let me know if that is what's expected.